### PR TITLE
fix(smart-contracts): use single function `updateLockConfig` to set lock key params

### PR DESCRIPTION
**Pull request by clemsos**
# Description

In order to emit events on all changes of the contract settings - while also keeping code size under control, we regroup three existing setters for lock key params (expiration duration, max keys, max keys per address) under a single `setLockConfig` signature

Impact on size 

```
 ····························|·············|················
 |  PublicLock               ·     24.347  ·       -0.064  │
 ·---------------------------|-------------|---------------·
```

relates to #9635 

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contracts/contracts/mixins/MixinKeys.sol [hunk 1]
```diff
@@ -654,41 +660,39 @@ contract MixinKeys is
...
+   function updateLockConfig(
+    uint _newExpirationDuration,
+    uint _maxKeysPerAcccount,
+    uint _maxNumberOfKeys
+  ) external {
      _onlyLockManager();
+     if(_maxKeysPerAcccount == 0) {
+       revert NULL_VALUE();
+     }
      if (_maxNumberOfKeys < _totalSupply) {
```
Comment (julien51) [inline smart-contracts/contracts/mixins/MixinKeys.sol:682]: I think we actually don't need to throw here...

File: smart-contracts/contracts/mixins/MixinKeys.sol [hunk 2]
```diff
@@ -654,41 +660,39 @@ contract MixinKeys is
...
+   * @dev _maxNumberOfKeys Can't be smaller than the existing supply 
    */
-  function setMaxNumberOfKeys (uint _maxNumberOfKeys) external {
+   function updateLockConfig(
+    uint _newExpirationDuration,
+    uint _maxKeysPerAcccount,
+    uint _maxNumberOfKeys
+  ) external {
      _onlyLockManager();
+     if(_maxKeysPerAcccount == 0) {
```
Comment (julien51) [inline smart-contracts/contracts/mixins/MixinKeys.sol:679]: Is that an issue though? I am not sure... It probably means nobody can purchase keys after this?

Comment (julien51) [reply smart-contracts/contracts/mixins/MixinKeys.sol:679]: We can maybe save some size by removing it

Comment (clemsos) [reply smart-contracts/contracts/mixins/MixinKeys.sol:685]: We can keep that for now as it wont be significant in terms of size but will make the config step more error prone

File: smart-contracts/contracts/interfaces/IPublicLock.sol [hunk 3]
```diff
@@ -74,13 +74,21 @@ interface IPublicLock
   function updateKeyPricing( uint _keyPrice, address _tokenAddress ) external;
 
   /**
+   * Set the maximum number of keys a specific address can use
```
Comment (julien51) [inline smart-contracts/contracts/interfaces/IPublicLock.sol:77]: Please update docstrings :)

File: smart-contracts/contracts/mixins/MixinKeys.sol [hunk 4]
```diff
@@ -660,41 +666,39 @@ contract MixinKeys is
   }
 
   /**
-   * @notice Change the maximum number of keys the lock can edit
+   * Set the maximum number of keys a specific address can use
```
Comment (julien51) [inline smart-contracts/contracts/mixins/MixinKeys.sol:669]: Update docstrings :)

**Review by julien51 — APPROVED** [review]

**Pull request merged**